### PR TITLE
chore: add more types, and remove unnecessary functions

### PR DESCRIPTION
**Pull request by itsyaasir**
# Issue ticket

Issue ticket link: <>

### Code conventions

- [x] I have performed a self-review of my code
- [x] Every new function is documented
- [x] Object names are auto explicative

### Security

- [x] The PR does not break APIs backward compatibility
- [x] The PR does not break the stable storage backward compatibility

### Testing

- [x] Every function is properly unit tested
- [x] I have added integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/did/src/transaction.rs [hunk 1]
```diff
@@ -237,50 +236,6 @@ pub struct Transaction {
     pub chain_id: Option<U256>,
 }
 
-impl Transaction {
```
Comment (ufoscout) [inline src/did/src/transaction.rs:240]: why did you remove them? They seem still useful to me

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/did/src/transaction.rs (hunk 1 above)
Comment (itsyaasir) [reply src/did/src/transaction.rs:240]: I am writing these functions in the evm which takes in required values since we are using the values from execution

Comment (ufoscout) [reply src/did/src/transaction.rs:240]: Shound't you add them to the StorableExecutionResult instead?

Comment (itsyaasir) [reply src/did/src/transaction.rs:240]: I thought about that it seems confusing and misrepresentation with what `StorableExecutionResult` does when compared to these methods.

Comment (itsyaasir) [reply src/did/src/transaction.rs:240]: But It can be added if you think it's all good!

**Review by ufoscout — APPROVED** [review]
